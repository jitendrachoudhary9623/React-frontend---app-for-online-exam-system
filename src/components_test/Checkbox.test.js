import React from 'react';
import { shallow } from 'enzyme';
import CheckBoxComponenet from '../components/admin/CreateTest/check-box.component';

const questionObj = {
    "id": 6,
    "op1": "Round",
    "op2": "Syllabus",
    "op3": "Rough",
    "op4": "Fragile",
    "ans_option": 3,
    "isselected": 0,
    "question": "Synonym of Coarse:"
};

function setUpComponenet(){
    return shallow(
        <CheckBoxComponenet 
           onChange = { () => {}}
           question={ questionObj }
           isChecked = { questionObj.isselected } />
      );
}

describe('CreateQuestionPaper', () => {
 it('should be defined', () => {
   expect(CheckBoxComponenet).toBeDefined();
 });
 it('should render correctly', () => {
   const tree = setUpComponenet();
   expect(tree).toMatchSnapshot();
 });
});