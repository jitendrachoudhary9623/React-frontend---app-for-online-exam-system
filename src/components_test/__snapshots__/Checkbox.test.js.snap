exports[`CreateQuestionPaper should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBoxComponenet
    isChecked={0}
    onChange={[Function]}
    question={
        Object {
            "ans_option": 3,
            "id": 6,
            "isselected": 0,
            "op1": "Round",
            "op2": "Syllabus",
            "op3": "Rough",
            "op4": "Fragile",
            "question": "Synonym of Coarse:",
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={0}
          className="filled-in"
          onChange={[Function]}
          type="checkbox" />,
        <span />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": 0,
          "className": "filled-in",
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={0}
            className="filled-in"
            onChange={[Function]}
            type="checkbox" />,
          <span />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": 0,
            "className": "filled-in",
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
